setup.sh

```diff
@@ -0,0 +1,178 @@
+        tput sgr0 # clear
+        echo 'See https://docs.joyent.com/public-cloud/api-access/docker'
+        exit 1
+    }
+    command -v json >/dev/null 2>&1 || {
+        echo
+        tput rev  # reverse
+        tput bold # bold
+        echo 'Error! JSON CLI tool is required, but does not appear to be installed.'
+        tput sgr0 # clear
+        echo 'See https://apidocs.joyent.com/cloudapi/#getting-started'
+        exit 1
+    }
+
+    command -v triton >/dev/null 2>&1 || {
+        echo
+        tput rev  # reverse
+        tput bold # bold
+        echo 'Error! Joyent Triton CLI is required, but does not appear to be installed.'
+        tput sgr0 # clear
+        echo 'See https://www.joyent.com/blog/introducing-the-triton-command-line-tool'
+        exit 1
+    }
+
+    # make sure Docker client is pointed to the same place as the Triton client
+    local docker_user=$(docker info 2>&1 | awk -F": " '/SDCAccount:/{print $2}')
+    local docker_dc=$(echo $DOCKER_HOST | awk -F"/" '{print $3}' | awk -F'.' '{print $1}')
+    TRITON_USER=$(triton profile get | awk -F": " '/account:/{print $2}')
+    TRITON_DC=$(triton profile get | awk -F"/" '/url:/{print $3}' | awk -F'.' '{print $1}')
+    TRITON_ACCOUNT=$(triton account get | awk -F": " '/id:/{print $2}')
+    if [ ! "$docker_user" = "$TRITON_USER" ] || [ ! "$docker_dc" = "$TRITON_DC" ]; then
+        echo
+        tput rev  # reverse
+        tput bold # bold
+        echo 'Error! The Triton CLI configuration does not match the Docker CLI configuration.'
+        tput sgr0 # clear
+        echo
+        echo "Docker user: ${docker_user}"
+        echo "Triton user: ${TRITON_USER}"
+        echo "Docker data center: ${docker_dc}"
+        echo "Triton data center: ${TRITON_DC}"
+        exit 1
+    fi
+
+    local triton_cns_enabled=$(triton account get | awk -F": " '/cns/{print $2}')
+    if [ ! "true" == "$triton_cns_enabled" ]; then
+        echo
+        tput rev  # reverse
+        tput bold # bold
+        echo 'Error! Triton CNS is required and not enabled.'
+        tput sgr0 # clear
+        echo
+        exit 1
+    fi
+
+    # setup environment file
+    if [ ! -f "_env" ]; then
+        echo '# Environment variables for MySQL service' > _env
+        echo 'MYSQL_USER=' >> _env
+        echo 'MYSQL_PASSWORD=' >> _env
+        echo 'MYSQL_REPL_USER=' >> _env
+        echo 'MYSQL_REPL_PASSWORD=' >> _env
+        echo 'MYSQL_DATABASE=' >> _env
+        echo >> _env
+
+        echo '# Environment variables for backups to Manta' >> _env
+        echo 'MANTA_BUCKET=' >> _env
+        echo 'MANTA_URL=' >> _env
+        echo 'MANTA_USER=' >> _env
+        echo 'MANTA_SUBUSER=' >> _env
+        echo 'MANTA_ROLE=' >> _env
+
+        # MANTA_KEY_ID must be the md5 formatted key fingerprint. A SHA256 will result in errors.
+        echo MANTA_KEY_ID=$(ssh-keygen -E md5 -lf ${MANTA_PUBLIC_KEY_PATH} | awk '{print substr($2,5)}') >> _env
```

Comment:
It occurs to me that rather than asking the user for their public key, we can just take the private key they're giving us and then fetching the signature from there. For example:

```
ssh-keygen -yl -E md5 -f ~/.ssh/timgross-joyent_id_rsa | awk '{print substr($2,5)}'
46:cc:9a:2d:66:83:1f:83:ce:01:74:5f:16:dc:b7:2d
```

That'd be one less argument to parse and one less place for user error to creep in.